adal/src/main/java/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -639,7 +642,16 @@ private AuthenticationResult processTokenResponse(HttpWebResponse webResponse, f
             }
 
             if (null != webResponse.getResponseHeaders().get(X_MS_CLITELEM) && !webResponse.getResponseHeaders().get(X_MS_CLITELEM).isEmpty()) {
-                httpEvent.setXMsCliTelemData(webResponse.getResponseHeaders().get(X_MS_CLITELEM).get(0));
+                final CliTelemInfo cliTelemInfo =
+                        TelemetryUtils.parseXMsCliTelemHeader(
```

Comment:
I would refactor this to call httpEvent.setXMsCliTelemData without any checks. And then http event's setter calls the parse routine. That way all telemetry logic is in telemetry classes.

Reply:
Hmm I'm not sure that's actually what we want, come to think of it.

The SPE ring info gets set in two places after it's parsed. The first place is on the `HttpEvent`, the second is on the `AuthenticationResult`. 

In order to hide the parsing data entirely inside of the `HttpEvent` class I'd need to put a getter on the event to get the spe ring data out (so that it can be set on the `AuthenticationResult`) -- are you OK with that?

There's nothing _techinically_ stopping me from making this change, but it would be the only `Event` class that has a getter on it and this feels a little strange.

To my eye, because this data gets set in two places within this method, it seems to make enough sense to do the parsing here. If that doesn't pass the sniff-test I'll go ahead and make this change and get the data back out of the `HttpEvent` with a getter.